skywalking-sniffer/skywalking-sdk-plugin/mongodb-plugin/src/main/java/com/a/eye/skywalking/plugin/mongodb/MongoDBReadBindingInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package com.a.eye.skywalking.plugin.mongodb;
+
+import com.a.eye.skywalking.api.context.ContextManager;
+import com.a.eye.skywalking.api.plugin.interceptor.EnhancedClassInstanceContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodInvokeContext;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import com.a.eye.skywalking.api.plugin.interceptor.enhance.MethodInterceptResult;
+import com.mongodb.ServerAddress;
+import com.mongodb.binding.ReadBinding;
+
+/**
+ * {@link MongoDBReadBindingInterceptor} record the host and port information from {@link EnhancedClassInstanceContext#context},
+ *
+ * @author baiyang
+ */
+public class MongoDBReadBindingInterceptor implements InstanceMethodsAroundInterceptor {
```

Comment:
@wu-sheng，Here is some code fragments of `MongoDB-Clinet`. The execute methods in `Mongo` class.

![mongo-exe](https://cloud.githubusercontent.com/assets/20091216/25138451/20d896c6-248d-11e7-92a7-99daba8bfea9.jpeg)

When `execute` method is executed, it is executed `getReadBinding ` or `getWriteBinding` method immediately.